shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/Address/CompactAddr.hs

```diff
@@ -62,6 +62,7 @@ propDecompactShelleyLazyAddr = do
       keyHash1 = unsafeGetHash . CA.decompactAddr . mangle . CA.compactAddr $ addr
    in pure $ keyHash0 == keyHash1
 
+-- | TODO This property test is failing to find a discrepancy that was found on mainnet.
```

Comment:
what's very confusing to me is that the `Arbitrary` instance for `Address` is generating byron addresses:

https://github.com/input-output-hk/cardano-ledger-specs/blob/ad92c1582c22471eb2f0c7195da82c29edb85d61/shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Serialisation/EraIndepGenerators.hs#L336

And the byron address generator looks to be generating redeem adrdesses:

https://github.com/input-output-hk/cardano-ledger-specs/blob/ad92c1582c22471eb2f0c7195da82c29edb85d61/byron/ledger/impl/test/Test/Cardano/Chain/Common/Gen.hs#L97